### ISSUE TITLE
Login Failed: Unknown Error

### ISSUE THREAD

**Issue by boompagirl** [cgeo/cgeo#8524]
**Describe the bug:**
When I try to authenticate with geocaching.com, I receive an error. Login Failed: Unknown Error. Please refer to our FAQ for guidance. I am able to logon to geocaching.com via chrome without issue.

**To Reproduce:**
1. Go to Settings
2. Click on Authorize c:geo
3. enter Username and Password
4. Click Check authentication.

**Actual behavior/state after performing these steps:**
Error message

**Expected behavior/state after performing these steps:**
Login and see nearby caches, etc.

**Version of c:geo used:**
2020.06.14

**Is the problem reproducible:**
Yes

**Screenshots:**
You may attach screenshots if applicable and helpful to explain your problem

**System information:**
```
--- System information ---
Device: SM-G950U (dreamqltesq, samsung)
Android version: 9
Android build: PPR1.180610.011.G950USQS7DTB3
c:geo version: 2020.06.14
Google Play services: enabled - 20.21.17 (100408-316502805)
Low power mode: inactive
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide caches: -
Hide waypoints: -
HW acceleration: enabled (default state)
System language: en_US
System date format: M/d/yy
Debug mode active: no
System internal c:geo dir: /data/user/0/cgeo.geocaching (27.3 GB free) internal
User storage c:geo dir: /storage/emulated/0/cgeo (27.3 GB free) external non-removable
Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (27.3 GB free) external non-removable
Database: /data/user/0/cgeo.geocaching/databases/data (156.0 KB) on system internal storage
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Not logged in (No Login information stored)
Geocaching.com date format: MM/dd/yyyy
Installed c:geo plugins:  none
--- End of system information ---
```

**Additional context:**
I recently performed a factory reset on my phone and prior to that had not had any issues with this app.

**Comment by Lineflyer** [comment]
In order to further analyze your problem, we need your help:
- Go to c:geo Menu - Settings - System und activate the "Debug Log" function
- Repeat the problematic function/action
- Go again into c:geo Menu - Settings - System and select "Create Logfile"
- Send this logfile to us to support@cgeo.org and mention this issue number
- After creating the logfile you should disable the "Debug Log" function

**Comment by boompagirl** [comment]
Sent. Thanks!

**Comment by Lineflyer** [comment]
--------- beginning of main
06-22 18:37:27.623 21014 22083 W cgeo    : [network--6] Login.login: No login information stored
06-22 18:37:48.316 21014 26524 W cgeo    : [network--9] Login.login: No login information stored
06-22 18:38:16.295 21014 21014 I cgeo    : [main] Requesting settings backup with settings manager
06-22 18:38:24.367 21014  8636 D cgeo    : [OkHttp] POST https://www.geocaching.com/account/logout
06-22 18:38:24.837 21014  8636 D cgeo    : [OkHttp] 403 [Forbidden] (469 ms) POST https://www.geocaching.com/account/logout (http/1.1)
06-22 18:38:24.854 21014  8636 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/signin
06-22 18:38:24.985 21014  8636 D cgeo    : [OkHttp] 200 (131 ms) GET https://www.geocaching.com/account/signin (http/1.1)
06-22 18:38:25.032 21014  8636 D cgeo    : [OkHttp] POST https://www.geocaching.com/account/signin
06-22 18:38:25.184 21014  8636 D cgeo    : [OkHttp] 200 (151 ms) POST https://www.geocaching.com/account/signin (http/1.1)
06-22 18:38:25.205 21014 27783 I cgeo    : [network--10] Failed to log in Geocaching.com as boompagirl for some unknown reason
06-22 18:38:25.218 21014 27783 I cgeo    : [network--10] Geocaching.com language already set to English
06-22 18:38:25.229 21014  8636 D cgeo    : [OkHttp] GET https://www.geocaching.com/account/signin
06-22 18:38:25.356 21014  8636 D cgeo    : [OkHttp] 200 (127 ms) GET https://www.geocaching.com/account/signin (http/1.1)
06-22 18:38:25.385 21014  8636 D cgeo    : [OkHttp] POST https://www.geocaching.com/account/signin
06-22 18:38:25.532 21014  8636 D cgeo    : [OkHttp] 200 (147 ms) POST https://www.geocaching.com/account/signin (http/1.1)
06-22 18:38:25.542 21014 27783 I cgeo    : [network--10] Failed to log in Geocaching.com as boompagirl for some unknown reason
06-22 18:38:35.335 21014 21014 I cgeo    : [main] Requesting settings backup with settings manager
06-22 18:38:35.370 21014 21014 D cgeo    : [main] LocationProvider: requesting most precise locations
06-22 18:38:35.384 21014 21061 W cgeo    : [network--1] Login.login: No login information stored
06-22 18:38:42.743 21014 21059 D cgeo    : [looper callbacks] LocationProvider: stopping location requests
06-22 18:39:08.984 21014 21014 I cgeo    : [main] Starting cgeo.geocaching 20200614 a.k.a 2020.06.14
06-22 18:39:08.985 21014 21059 D cgeo    : [looper callbacks] LocationProvider: requesting most precise locations
06-22 18:39:09.033 21014 21088 W cgeo    : [network--2] Login.login: No login information stored

**Comment by Lineflyer** [comment]
This is the still unresolved login issue some users experience, which is most of the time magically solved by reinstallation.
Thanks to the new logging file feature we do now have at least a debug log.

Does any of the developers have an idea?

**Comment by Lineflyer** [comment]
Link to #5963 and #5892

**Comment by Lineflyer** [comment]
Does anyone have a clue or idea what might be going on here?
From the log I don't see the reason for failing here.

Else: Is is possible to implement better debug logging here to have a future possibility of debugging such kind of problems.

There are still more or less frequently people around on our support channels with this problem, which is most of the time solved by reinstalling or resetting c:geo and I still have no clue why this problem appears after all.

**Comment by Lineflyer** [comment]
This has turned out to be a credential problem but caused by our wrong error message "unknown reason" it was not clear. The detection of incorrect credentials has been fixed meanwhile and the fix will be included in the next c:geo release.

**Comment by bekuno** [comment]
fixed by #8530
